subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperationWorker.java

```diff
@@ -16,16 +16,16 @@
 
 package org.gradle.internal.operations;
 
-import org.gradle.api.Action;
-
 /**
- * A worker that can run build operations of type {@link T}.
+ * A worker that can run build operations.
  *
  * Implementations must be thread-safe.
  */
-public interface BuildOperationWorker<T extends BuildOperation> extends Action<T> {
+public interface BuildOperationWorker<O extends BuildOperation> {
     /**
      * Returns a human consumable name for this tool.
      */
     String getDisplayName();
+
+    void execute(O buildOperation);
```

Comment:
Ok I got that wrong.

I think the confusion was created for me when `Executor` and `Processor` were initially combined. The _Worker_ with which the queue factory is configured is now always supposed to be "a worker" that works through the executor (firing build operation events etc.). As such, there should never be a use case were a custom worker (i.e. implementation of `BuildOperationWorker`) is directly passed to the queue factory.

This actually became clearer now as I added operation context as argument to `BuildOperationWorker`.

I use a separate interface now for initialising the queue factory - `BuildOperationQueue.QueueWorker`.

The previous behavior should now be restored.